### PR TITLE
Feature/healthcheck

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

- Replaced healthcheck with dp-healthcheck
- Migraged datasetAPI and importAPI functionality to dp-api-clients-go, keeping a wrapper in this repository for error handling.
- Fixed graceful shutdown and refactored concurrency handling with waitGroups. Note that I had to shutdown the healthcheck before anything else could start shutting down, to prevent race conditions.
- Log Kafka errors, instead of terminate the service.
- Replaced go-ns/log with log.go
- Replaced go-ns/kafka with dp-kafka
- converted to go module, with go version 1.13, and fixed CI scripts and yml accordingly

### How to review

- Make sure the code changes make sense
- You can try the `/health` endpoint, and see the Checkers results in different scenarios
- If Kafka is not running, the service should not terminate.
- Unit tests should pass

### Who can review

Anyone.

### CODE REVIEW

File: cmd/input-file-available-producer/main.go [hunk 1]
```diff
@@ -17,11 +19,12 @@ var kafkaHost = flag.String("kafka", "localhost:9092", "")
 func main() {
 
 	flag.Parse()
+	ctx := context.Background()
 
 	var brokers []string
 	brokers = append(brokers, *kafkaHost)
 
-	producer, _ := kafka.NewProducer(brokers, *topic, int(2000000))
+	producer, _ := kafka.NewProducer(ctx, brokers, *topic, int(2000000), kafka.CreateProducerChannels())
```
Comment (nshumoogum) [inline cmd/input-file-available-producer/main.go]: If you just handle the error here and call os.Exit(1) - this package might be removed at a latter date

File: cmd/observations-inserted-producer/main.go [hunk 2]
```diff
@@ -16,11 +18,12 @@ var kafkaHost = flag.String("kafka", "localhost:9092", "")
 func main() {
 
 	flag.Parse()
+	ctx := context.Background()
 
 	var brokers []string
 	brokers = append(brokers, *kafkaHost)
 
-	producer, _ := kafka.NewProducer(brokers, *topic, int(2000000))
+	producer, _ := kafka.NewProducer(ctx, brokers, *topic, int(2000000), kafka.CreateProducerChannels())
```
Comment (nshumoogum) [inline cmd/observations-inserted-producer/main.go]: Same again

File: go.sum [hunk 3]
```diff
@@ -0,0 +1,149 @@
+github.com/ONSdigital/dp-api-clients-go v1.6.1-0.20200304105619-e96c4a4fa265 h1:yFKqrI6X//y6BCgexcplwG7xy9Y4QHZMjUoRVceHvOM=
```
Comment (nshumoogum) [inline go.sum]: If you just run `go mod tidy` once all dependencies have been updated

File: cmd/dp-import-tracker/main.go [hunk 4]
```diff
@@ -395,97 +406,145 @@ func produceDataImportCompleteEvents(
...
 		cfg.Brokers,
 		cfg.NewInstanceTopic,
 		cfg.NewInstanceConsumerGroup,
-		kafka.OffsetNewest)
+		kafka.OffsetNewest,
+		true,
+		kafka.CreateConsumerGroupChannels(true))
 	if err != nil {
-		logFatal("could not obtain consumer", err, log.Data{"topic": cfg.NewInstanceTopic})
+		logFatal(mainContext, "could not obtain consumer", err, log.Data{"topic": cfg.NewInstanceTopic})
```
[the remark this answers is not in the document]
Comment (DavidSubiros) [reply cmd/dp-import-tracker/main.go]: At this point, we only return an error if there is a channel validation error. If the Kafka broker(s) cannot be contacted we don't return an error, but the consumer/producer is not initialised. The health check will initialise it once the brokers are reachable.

File: cmd/input-file-available-producer/main.go (hunk 1 above)
Comment (DavidSubiros) [reply cmd/input-file-available-producer/main.go]: Also, I will replace the code below `panic(err)` for `os.Exist(1)`

**Pull request merged**